File: agent/sdk/python/examples/offline_receipt_verification.py
from layerx_sdk import (
    AuthorizedReceiptBatch,
    LocalSignatureVerifier,
    ReceiptVerification,
    verify_receipt,
)


def offline_receipt_verification(
    canonical_receipt: bytes,
    authorized_batch: AuthorizedReceiptBatch,
    signatures: LocalSignatureVerifier,
) -> ReceiptVerification:
    return verify_receipt(canonical_receipt, authorized_batch, signatures)

File: agent/sdk/typescript/examples/offline-receipt-verification.ts
import {
  verifyReceipt,
  type AuthorizedReceiptBatch,
  type ReceiptVerification,
} from "../src/index.js";

export function offlineReceiptVerification(
  canonicalReceipt: Uint8Array,
  authorizedBatch: AuthorizedReceiptBatch,
): Promise<ReceiptVerification> {
  return verifyReceipt(canonicalReceipt, authorizedBatch);
}
